opcua/common/connection.py

```diff
@@ -299,7 +299,14 @@ def receive_from_header_and_body(self, header, body):
         if header.MessageType == ua.MessageType.SecureOpen:
             data = body.copy(header.body_size)
             security_header = struct_from_binary(ua.AsymmetricAlgorithmHeader, data)
-            self.select_policy(security_header.SecurityPolicyURI, security_header.SenderCertificate)
+
+            if self.is_open():
+                # In case the channel is already open we don't want to call
+                # select_policy. Since it will break the Secure channel renewal
+                pass
+            else:
```

Comment:
better write:

if not self.is_open():
   do something

instead of suing pass. but keep the comment

Reply:
done